symengine/eval_mpfr.cpp

```diff
@@ -216,7 +216,11 @@ class EvalMPFRVisitor : public BaseVisitor<EvalMPFRVisitor> {
         if (x.__eq__(*pi)) {
             mpfr_const_pi(result_, rnd_);
         } else if (x.__eq__(*E)) {
-            mpfr_const_euler(result_, rnd_);
+            mpfr_t one_;
+            mpfr_init2(one_, mpfr_get_prec(result_));
+            mpfr_set_d(one_, 1, rnd_);
```

Comment:
Precision of double is 53 whereas the precision of `result_` can be much larger than that. I'm not sure of the internals of MPFR and whether having 1 as a double matters, but it's always better to use exact numbers.